backend/types/types.go

```diff
@@ -4,17 +4,20 @@ import (
 	"github.com/atlassian/gostatsd/types"
 
 	"github.com/spf13/viper"
+	"golang.org/x/net/context"
 )
 
-// Factory is a function that returns a MetricSender.
-type Factory func(*viper.Viper) (MetricSender, error)
+// Factory is a function that returns a Backend.
+type Factory func(*viper.Viper) (Backend, error)
 
-// MetricSender represents a backend.
-type MetricSender interface {
+// Backend represents a backend.
+type Backend interface {
 	// BackendName returns the name of the backend.
 	BackendName() string
 	// SampleConfig returns the sample config for the backend.
 	SampleConfig() string
 	// SendMetrics flushes the metrics to the backend.
-	SendMetrics(*types.MetricMap) error
+	SendMetrics(context.Context, *types.MetricMap) error
```

Comment:
Yes, so if we need to cancel the send, we could. We will pass context to http library/etc when it is supported (should be in 1.7 I think?).